generators/csharp/sdk/src/SdkGeneratorCli.ts

```diff
@@ -145,6 +146,14 @@ export class SdkGeneratorCLI extends AbstractCsharpGeneratorCli<SdkCustomConfigS
         const baseApiException = new BaseApiExceptionGenerator(context);
         context.project.addSourceFiles(baseApiException.generate());
 
+        if (context.ir.auth.schemes[0] != null && context.ir.auth.schemes[0].type === "oauth") {
+            const oauthTokenProvider = new OauthTokenProviderGenerator({
+                context,
+                scheme: context.ir.auth.schemes[0]
+            });
+            context.project.addSourceFiles(oauthTokenProvider.generate());
+        }
```

Comment:
Should we generate the provider if _any_ of the auth schemes specify `oauth`? I imagine there will be some customers that include multiple, but maybe list `bearer` first.

Reply:
The behavior is kind of off here across generators -- ie most things only work with the first auth scheme

Reply:
Decision: Okay to do first auth scheme but there is an IR change that will make this a union forcing generators to handle all, any, or one.